packages/blitz-rpc/src/data-client/rpc.ts

```diff
@@ -103,12 +103,10 @@ export function __internal_buildRpcClient({
     })
 
     if (httpMethod === "GET") {
-      fullRoutePath +=
-        "?" +
-        new URLSearchParams({
-          params: stringify(serialized.json),
-          meta: stringify(serialized.meta),
-        })
+      const url = new URL(fullRoutePath, window.location.origin)
+      url.searchParams.set("params", stringify(serialized.json))
+      url.searchParams.set("meta", stringify(serialized.meta))
+      fullRoutePath = url.toString()
```

Comment:
This is not really better, because it overwrites the original `fullRoutePath`

Probably you should change `fullRoutePath` up above to be `routePath` and then here do like this. So `routePath` is never overwritten (change to const)

```suggestion
      const url = new URL(routePath, window.location.origin)
      url.searchParams.set("params", stringify(serialized.json))
      url.searchParams.set("meta", stringify(serialized.meta))
      fullRoutePath = url.toString()
```

Reply:
@flybayer I made the requested changes